test/sslapitest.c

```diff
@@ -13811,6 +13811,320 @@ static int test_ssl_trace(void)
 }
 #endif
 
+
+static int test_ssl_conf_CertValidation(SSL_CONF_CTX *confctx,
+					SSL_CTX *ssl_ctx,
+					SSL *ssl)
+{
+    int ret = 0;
+    X509_VERIFY_PARAM *param;
+    unsigned long flags;
+
+    if (ssl_ctx) {
+	param = SSL_CTX_get0_param(ssl_ctx);
+    } else if (ssl) {
+	param = SSL_get0_param(ssl);
+    } else {
+	goto end;
+    }
+
+    /* Verify that we can set and clear flags using "CertValidation" */
+    if (!TEST_int_eq(SSL_CONF_cmd(confctx, "CertValidation", "UseCheckTime"), 2))
+	goto end;
+
+    flags = X509_VERIFY_PARAM_get_flags(param);
+    if (!(flags & X509_V_FLAG_USE_CHECK_TIME))
+	goto end;
+
+    if (!TEST_int_eq(SSL_CONF_cmd(confctx, "CertValidation", "-UseCheckTime"), 2))
+	goto end;
+
+    flags = X509_VERIFY_PARAM_get_flags(param);
+    if (flags & X509_V_FLAG_USE_CHECK_TIME)
+	goto end;
+
+    ret = 1;
+end:
+    return ret;
+}
+
+static int test_ssl_conf_SetValidFlags(SSL_CONF_CTX *confctx,
+				       SSL_CTX *ssl_ctx,
+				       SSL *ssl)
+{
+    int ret = 0;
+    X509_VERIFY_PARAM *param;
+    unsigned int flags;
+
+    if (ssl_ctx) {
+	param = SSL_CTX_get0_param(ssl_ctx);
+    } else if (ssl) {
+	param = SSL_get0_param(ssl);
+    } else {
+	goto end;
+    }
+
+    /* Verify that we can set and clear flags using "SetValidFlags" */
+    if (!TEST_int_eq(SSL_CONF_cmd(confctx, "SetValidFlags", "AlwaysCheckSubject"), 2))
+        goto end;
+
+    flags = X509_VERIFY_PARAM_get_hostflags(param);
+    if (!(flags & X509_CHECK_FLAG_ALWAYS_CHECK_SUBJECT)) {
```

Comment:
Style nit:
```suggestion
    if ((flags & X509_CHECK_FLAG_ALWAYS_CHECK_SUBJECT) == 0) {
```

Same style nit applies in function above